osu.Game/Online/Chat/DrawableLinkCompiler.cs

```diff
@@ -15,13 +18,17 @@ namespace osu.Game.Online.Chat
     /// <summary>
     /// An invisible drawable that brings multiple <see cref="Drawable"/> pieces together to form a consumable clickable link.
     /// </summary>
-    public class DrawableLinkCompiler : OsuHoverContainer
+    public class DrawableLinkCompiler : OsuHoverContainer, IHasContextMenu
     {
         /// <summary>
         /// Each word part of a chat link (split for word-wrap support).
         /// </summary>
         public List<Drawable> Parts;
 
+        public OsuGame Game;
+
+        public LinkDetails Link;
```

Comment:
public fields are generally a no-go here, `Parts` is more an exception than a rule. `OsuGame` should be DI'd, and `LinkDetails` should be `{ get; private set; }` if exposition is deemed absolutely necessary